[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44074 [quant][graphmode][fx] Support quantization for custom module
* **#43990 [quant][eagermode][refactor] Add set/get method for quantization and fusion mappings**

Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_mapping
- get_static_quant_module_mappings
- get_static_quant_module_class

- register_dynamic_quant_module_mapping
- get_dynamic_quant_module_mappings

- register_qat_module_mapping
- get_qat_module_mappings

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_mapping(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

jerryzh168 committed:
[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings

Summary:
Allow user to register custom quantization and fusion patterns

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
nice! any chance we could add an example usage of this API to register a custom pattern to the PR summary?  would make it easier to review

z-a-f commented on torch/quantization/_numeric_suite.py:
```diff
@@ -437,7 +422,7 @@ def compare_model_outputs(
     *data,
     Logger=OutputLogger,
-    white_list=DEFAULT_NUMERIC_SUITE_COMPARE_MODEL_OUTPUT_WHITE_LIST,
+    white_list=None
```
Should we change the `white_list` while we are at it?

jerryzh168 replied:
i'm afraid this is bc breaking, would be better to change all at once

z-a-f reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 commented:
> nice! any chance we could add an example usage of this API to register a custom pattern to the PR summary? would make it easier to review

done

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

vkuzo commented on torch/quantization/quantization_mappings.py:
```diff
@@ -0,0 +1,176 @@
+    F.layer_norm: torch._ops.ops.quantized.layer_norm,
+}
+
+def register_static_quant_module_class(module_class, static_quant_module_class):
```
optional, how about something like this to improve readability

```
def register_static_quant_module_swap(fp_source_module_class, static_quant_target_module_class):
    """Registers a module swap of {fp_source_module_class} to {static_quant_target_module_class} 
    during static quant convert step.  static_quant_target_module_class must have from_float 
    defined as a class method."""
```

jerryzh168 replied:
sure, will change to `register_static_quant_module_mapping`

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_class
- get_static_quant_module_mapping
- get_static_quant_module_class

- register_dynamic_quant_module_class
- get_dynamic_quant_module_mapping

- register_qat_module_class
- get_qat_module_mapping

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_class(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_class(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][eagermode][refactor] Add set/get method for quantization and fusion mappings"


Summary:
Allow user to register custom quantization and fusion patterns

Custom quantization API:
- register_static_quant_module_mapping
- get_static_quant_module_mappings
- get_static_quant_module_class

- register_dynamic_quant_module_mapping
- get_dynamic_quant_module_mappings

- register_qat_module_mapping
- get_qat_module_mappings

for fx graph mode quantization
- register_quantized_operator
- get_quantized_operator

Custom fusion API:
- register_fuser_method
- get_fuser_method

Example:

```
register_static_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.Conv2d)
register_dynamic_quant_module_mapping(torch.nn.Conv2d, torch.nn.quantized.dynamic.Conv2d)
register_qat_module_mapping(torch.nn.Conv2d, torch.nn.qat.Conv2d)
register_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d), fuse_conv_bn)

>> get_static_quant_module_class(torch.nn.Conv2d)
torch.nn.quantized.Conv2d

>> get_fuser_method((torch.nn.Conv2d, torch.nn.BatchNorm2d))
fuser_conv_bn
```

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23485344](https://our.internmc.facebook.com/intern/diff/D23485344)

[ghstack-poisoned]
